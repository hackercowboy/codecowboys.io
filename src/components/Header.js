import React from 'react';
import PropTypes from 'prop-types';
import Head from 'next/head';

function Header({ title, description, children }) {
  return (
    <header>
      <Head>
        <title>
          {title ? `${title} | codecowboys.io` : 'codecowboys.io'}
        </title>
        {description && (<meta name="description" content={description} />)}
      </Head>
      <div className="container header-wrapper">
        <a href="/" className="header-brand">
          <img src="/logo.svg" alt="Codecowboys.io logo" style={{ width: '40px', height: '40px' }} />
          <div>codecowboys</div>
        </a>
        {children}
      </div>
    </header>
  );
}

Header.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  children: PropTypes.node,
};

Header.defaultProps = {
  title: undefined,
  description: undefined,
  children: undefined,
};

export default Header;
