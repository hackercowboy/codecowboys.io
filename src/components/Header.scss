@import '../styles/colors';

header {
  background: rgba(255, 255, 255, 0.9);
  border-bottom: solid 1px $border-color;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1000;

  .header-wrapper {
    display: flex;
  }

  .header-brand {
    display: flex;
    font-size: 30px;
    font-weight: 800;
    line-height: 40px;
    padding: 8px 0;
    text-decoration: none;

    img {
      margin-right: 10px;
      margin-top: 2px;
    }
  }
}
