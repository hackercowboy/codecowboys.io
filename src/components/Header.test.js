import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

describe('<Header/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Header><div /></Header>);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render with title and description', () => {
    const wrapper = shallow(<Header title="Tile" description="Description"><div /></Header>);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
