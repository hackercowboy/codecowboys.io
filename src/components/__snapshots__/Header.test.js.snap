// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render 1`] = `
"<header>
  <Head>
    <title>
      codecowboys.io
    </title>
  </Head>
  <div className=\\"container header-wrapper\\">
    <a href=\\"/\\" className=\\"header-brand\\">
      <img src=\\"/logo.svg\\" alt=\\"Codecowboys.io logo\\" style={{...}} />
      <div>
        codecowboys
      </div>
    </a>
    <div />
  </div>
</header>"
`;

exports[`<Header/> should render with title and description 1`] = `
"<header>
  <Head>
    <title>
      Tile | codecowboys.io
    </title>
    <meta name=\\"description\\" content=\\"Description\\" />
  </Head>
  <div className=\\"container header-wrapper\\">
    <a href=\\"/\\" className=\\"header-brand\\">
      <img src=\\"/logo.svg\\" alt=\\"Codecowboys.io logo\\" style={{...}} />
      <div>
        codecowboys
      </div>
    </a>
    <div />
  </div>
</header>"
`;
